pkg/controllers/nodeclass/suite_test.go

```diff
@@ -889,6 +890,118 @@ var _ = Describe("NodeClassController", func() {
 			nodeClass = ExpectExists(ctx, env.Client, nodeClass)
 			Expect(nodeClass.Annotations[v1beta1.AnnotationEC2NodeClassHash]).To(Equal(expectedHash))
 		})
+		It("should update ec2nodeclass hash version when the ec2nodeclass hash version is out of sync with the controller", func() {
+			nodeClass.Annotations = map[string]string{v1beta1.AnnotationEC2NodeClassHash: "test"}
+			ExpectApplied(ctx, env.Client, nodeClass)
+
+			ExpectReconcileSucceeded(ctx, nodeClassController, client.ObjectKeyFromObject(nodeClass))
+			nodeClass = ExpectExists(ctx, env.Client, nodeClass)
+
+			expectedHash := nodeClass.Hash()
+			Expect(nodeClass.Annotations).To(HaveKeyWithValue(v1beta1.AnnotationEC2NodeClassHash, expectedHash))
+			Expect(nodeClass.Annotations).To(HaveKeyWithValue(v1beta1.AnnotationEC2NodeClassHashVersion, v1beta1.EC2NodeClassHashVersion))
+		})
+		It("should update ec2nodeclass hash versions on all nodeclaims when the hash versions don't match", func() {
```

Comment:
In this testcase, is the nodeclass hash version set to actually reflect a value that's different than the current hash version? Or is this just working because the hash version doesn't exist yet and that still counts as the version being different?

Reply:
Updated the test to better reflect the values